scripts.py

```diff
@@ -1269,77 +1267,114 @@ def make_query(self, _db, identifier_type, identifiers, log=None):
+                        computed_sort_name=computed_sort_name_local_new, log=log)
+                    output = "%s|\t%s|\t%s|\t%s|\t%s|\tcomplain" % (contributor.id, contributor.sort_name, contributor.display_name, computed_sort_name_local_new, match_ratio)
+                    print output.encode("utf8")
+                else:
+                    # we can fix it!
+                    output = "%s|\t%s|\t%s|\t%s|\t%s|\tfix" % (contributor.id, contributor.sort_name, contributor.display_name, computed_sort_name_local_new, match_ratio)
+                    print output.encode("utf8")
+
+                    contributor.sort_name = computed_sort_name_local_new
 
-        computed_sort_name_local = display_name_to_sort_name(contributor.display_name, advanced=True)
-        success = cls.check_sort_name(computed_sort_name_local, contributor, edition, DataSource.INTERNAL_PROCESSING, CoverageRecord.REPAIR_SORT_NAME_OPERATION)
-        if not success:
-            output = "Contributor[%s]: contributor_sort_name=%s, contributor_display_name=%s: INTERNAL MISMATCH" % (contributor.id, contributor.sort_name, contributor.display_name)
-            print output.encode("utf8")
+                    # also change edition.sort_author, if the author was primary
+                    edition_contributors = contribution.edition.author_contributors
+                    if (len(edition_contributors) > 0 and (edition_contributors[0].id == contributor.id)):
+                        contribution.edition.sort_author = computed_sort_name_local_new
 
 
     @classmethod
-    def check_sort_name(cls, computed_sort_name, contributor, edition, data_source, operation):
+    def register_problem(cls, source, contribution, contributor, computed_sort_name, log=None):
+
         success = True
-        if computed_sort_name.strip().lower() != contributor.sort_name.strip().lower():
-            record, is_new = CoverageRecord.add_for(
-                edition=edition, data_source=DataSource.VIAF, operation=CoverageRecord.REPAIR_SORT_NAME_OPERATION,
-                status=CoverageRecord.TRANSIENT_FAILURE
-            )
-            # NOTE:  not the best way -- storing id info in the message field, but 
-            # perhaps better than making CoverageRecords cover Contributor-type objects.
-            record.exception = contributor.id
+
+        detail = "Contributor[id=%s].sort_name is oddly different from computed_sort_name, human intervention required." % contributor.id
+        pool = contribution.edition.is_presentation_for
+        complaint_type = "http://librarysimplified.org/terms/problem/wrong-author";
+        try:
+            complaint, is_new = Complaint.register(pool, complaint_type, source, detail)
+        except ValueError, e:
+            # log and move on, don't stop run
+            log.error("Error registering complaint: %r", contributor, exc_info=e)
+            print("Error registering complaint: %r", contributor)
```

Comment:
Yeah, this should be a log statement but you don't need the log from the instance, you can just do a `logging.getlogger` call.